app/Domain/Newsletter/Controllers/NewsletterController.php

```diff
@@ -5,22 +5,30 @@
 namespace App\Domain\Newsletter\Controllers;
 
 use App\Http\Controllers\Controller;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Validator;
 use Mailjet\Client;
 use Mailjet\Resources;
-use Mailjet\Response;
 
 final class NewsletterController extends Controller
 {
-    public function handle(string $email): Response
+    public function handle(string $email): JsonResponse
     {
-        if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
-            throw new \InvalidArgumentException('Invalid email address');
+        $validator = Validator::make(
+            ['email' => $email],
+            ['email' => 'required|email']
+        );
+        if ($validator->fails()) {
```

Comment:
I think you can do `$validator->validate();`

Reply:
But that throws a `ValidationException` which you need to catch?

Reply:
I think that is also how the `rules` method works on `FormRequest`

Reply:
Indeed, there doesn't really seem an elegant solution to do this without a `redirect`.